neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/internal/LazyStoreMap.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2013-2018 Atlanmod, Inria, LS2N, and IMT Nantes.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v2.0 which accompanies
+ * this distribution, and is available at https://www.eclipse.org/legal/epl-2.0/
+ */
+
+package fr.inria.atlanmod.neoemf.core.internal;
+
+import fr.inria.atlanmod.neoemf.core.PersistentEObject;
+import fr.inria.atlanmod.neoemf.data.store.Store;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.util.EcoreEMap;
+
+import java.util.List;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * A {@link java.util.Map} representing a multi-valued feature which behaves as a proxy and that delegates its
+ * operations to the associated {@link Store}.
+ *
+ * @param <K> the type of keys maintained by this map
+ * @param <V> the type of mapped values
+ *
+ * @see PersistentEObject#eStore()
+ */
+@ParametersAreNonnullByDefault
+public class LazyStoreMap<K, V> extends EcoreEMap<K, V> {
+
+    @SuppressWarnings("JavaDoc")
+    private static final long serialVersionUID = 9173875843551606055L;
+
+    /**
+     * Constructs a {@code LazyStoreMap} with the given {@code owner} and {@code feature}.
+     *
+     * @param owner   the owner the {@code feature}
+     * @param feature the feature associated with this map
+     */
+    public LazyStoreMap(PersistentEObject owner, EStructuralFeature feature) {
+        super(EClass.class.cast(feature.getEType()), Entry.class, null);
+
+        this.delegateEList = new EntryList(owner, feature);
+        this.size = delegateEList.size();
+    }
+
+    /**
+     * A {@link List} that holds entries of this map.
+     */
+    @ParametersAreNonnullByDefault
+    private class EntryList extends LazyStoreList<Entry<K, V>> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This class has 12 parents which is greater than 5 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)